astropy/units/structured.py

```diff
@@ -121,7 +121,7 @@ class StructuredUnit:
     Structured units share most methods with regular units::
 
         >>> su.physical_type
-        ((PhysicalType('length'), PhysicalType({'speed', 'velocity'})), PhysicalType('time'))
+        astropy.units.structured.Structure((astropy.units.structured.Structure((PhysicalType('length'), PhysicalType({'speed', 'velocity'})), dtype=[('f0', 'O'), ('f1', 'O')]), PhysicalType('time')), dtype=[('f0', 'O'), ('f1', 'O')])
```

Comment:
How does @namurphy and @nstarman feel about this?

Reply:
Agreed! I think the current ``((PhysicalType('length'), PhysicalType({'speed', 'velocity'})), PhysicalType('time'))`` should be the behaviour of `__str__`.
For `__repr__` we at least don't want to show the module `structured`.

```suggestion
        astropy.units.Structure((astropy.units.Structure((PhysicalType('length'), PhysicalType({'speed', 'velocity'})), dtype=[('f0', 'O'), ('f1', 'O')]), PhysicalType('time')), dtype=[('f0', 'O'), ('f1', 'O')])
```

Reply:
Agreed, that is the minimal change. But coming back to this fairly substantial PR, it is probably best to do that afterwards - I raised a new issue at #15793.

Reply:
Totally fine by me. I can review this PR in a few days, if you're ready @mhvk

Reply:
Thanks! This PR is basically waiting for numpy 2.0 to be released...  I'm just keeping it up to date once in a while until that happens.